mcs/class/referencesource/System.Data.Entity/System/Data/Objects/ELinq/MethodCallTranslator.cs

```diff
@@ -1363,6 +1378,10 @@ internal TrimTranslator()
                 private static IEnumerable<MethodInfo> GetMethods()
                 {
                     yield return typeof(String).GetMethod("Trim", BindingFlags.Public | BindingFlags.Instance, null, new Type[] { typeof(Char[]) }, null);
+#if MONO
+                    yield return typeof(String).GetMethod("Trim", BindingFlags.Public | BindingFlags.Instance, null, Type.EmptyTypes, null);
```

Comment:
This will still fail because the expects argument to be an array, see `if (!IsEmptyArray(call.Arguments[0]))` check in base class

Reply:
It will fail a bit earlier, during method translators initialization because `CanonicalFunctionDefaultTranslator` already contains this:

https://github.com/mono/mono/blob/9177dc18a0099251dcdc751939f1fec39c23b947/mcs/class/referencesource/System.Data.Entity/System/Data/Objects/ELinq/MethodCallTranslator.cs#L844